CDCgov/prime-reportstream
# Revert Delivery endpoint implementation

This PR re-implements `rest-hooks` and the old History API due to bugs in Delivery API back-end code.

Test Steps:
1. Ensure `/daily-data` and `/details` load as a non-Admin receiver
2. Ensure they load as an admin user mocked as a receiver

<img width="1154" alt="Screen Shot 2022-08-23 at 10 33 10 AM" src="https://user-images.githubusercontent.com/4022304/186195416-e25ad121-ecdc-432b-9dfe-829cc14b031c.png">
<img width="1156" alt="Screen Shot 2022-08-23 at 11 41 35 AM" src="https://user-images.githubusercontent.com/4022304/186201958-1189345a-7dba-40f1-b8df-99aa7fc67b44.png">


## Changes
- Comments out test due to rest-hooks testing infra issue
- Removes usage of new deliveries code for old rest-hooks usage
- Some component prop types extended to take `ReportResource` again

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
